### PR TITLE
feat: Add delta table `wholesale_results`

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## References

* https://app.zenhub.com/workspaces/mandalorian-5fd22a1a59e4740018f5ab5a/issues/gh/energinet-datahub/green-energy-hub/470

### CODE REVIEW

File: source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql [hunk 1]
```diff
@@ -0,0 +1,38 @@
+CREATE TABLE IF NOT EXISTS {OUTPUT_DATABASE_NAME}.wholesale_results
+(
+    batch_id STRING NOT NULL,
+    process_type STRING NOT NULL,
+    execution_time_start TIMESTAMP NOT NULL,
+    
+    result_id STRING NOT NULL,
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql:7]: also rename in energy_results?

File: source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    grid_area STRING NOT NULL,
+    energy_supplier_id STRING,
+    -- Energy quantity in for the given observation time and duration as defined by `resolution`.
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql:11]:
```suggestion
    -- Energy quantity for the given observation time and duration as defined by `resolution`.
```

File: source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    quantity DECIMAL(18, 3),
+    quantity_unit STRING NOT NULL,
+    quantity_quality STRING NOT NULL,
+    -- The time when the energy was consumed/produced/exchanged
+    time TIMESTAMP NOT NULL,
+    resolution STRING NOT NULL,
+
+    -- Null when monthly sum
+    metering_point_type STRING,
+    settlement_method STRING NOT NULL,
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql:24]: is NULL for production

File: source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    energy_currency STRING NOT NULL,
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql:26]: should we leave this one out?

File: source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    price DECIMAL(18, 3),
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql:27]: 6 decimals

File: source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    amount DECIMAL(18, 3) NOT NULL,
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql:28]: 6 decimals

File: source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql (hunk 4 above)
Comment (JesperGomes) [reply source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql:26]: Or maybe it’s nice to keep it? Your thoughts?

Comment (BjarkeMeier) [reply source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql:27]: It's required according to the notes I received from you.

File: source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql (hunk 1 above)
Comment (BjarkeMeier) [reply source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql:7]: I think we should probably pay attention to how many refactorings/renamings we do in the existing table in order to not obstruct the DNA team. Let's have a talk about naming.

File: source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql (hunk 5 above)
Comment (BjarkeMeier) [reply source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql]: Where can I see that?

File: source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql (hunk 4 above)
Comment (BjarkeMeier) [reply source/databricks/calculation-engine/package/datamigration/migration_scripts/202308221500_Add_wholesale_results_table.sql]: Removed

**Review by JesperGomes — DISMISSED** [review]

**Review by JesperGomes — APPROVED** [review]

**Pull request merged**